sql/core/src/main/scala/org/apache/spark/sql/Column.scala

```diff
@@ -1001,17 +1001,11 @@ class Column(val expr: Expression) extends Logging {
    *   df.select($"colA".name("colB"))
    * }}}
    *
-   * If the current column has metadata associated with it, this metadata will be propagated
-   * to the new column.  If this not desired, use `as` with explicitly empty metadata.
-   *
    * @group expr_ops
    * @since 2.0.0
    */
   def name(alias: String): Column = withExpr {
-    expr match {
-      case ne: NamedExpression => Alias(expr, alias)(explicitMetadata = Some(ne.metadata))
-      case other => Alias(other, alias)()
-    }
+    Alias(expr, alias)()
```

Comment:
This function introduced in #11908, the change here will also influence `def alias(alias: String): Column` and `def as(alias: String): Column`, do you check all the test cases related?

Reply:
Yes, I have successfully run all relevant UT test. 
(I‘m sorry that we have a long holiday recently, so the reply is late.

Reply:
Related tests maybe not enough, as your change, `as` function has a different behavior, `metadata` no longer pass to `explicitMetadata`. My suggestion:
1. For safety, run all test, not only the related.
2. Do the fix just for SS scenario.